src/database/queries/userDataProviderPermissions/insertOrUpdateOne.sql

```diff
@@ -2,13 +2,15 @@ INSERT INTO user_data_provider_permissions (
   user_keycloak_user_id,
   permission,
   data_provider_short_code,
-  created_by
+  created_by,
+  not_after
 ) VALUES (
   :userKeycloakUserId,
   :permission::permission_t,
   :dataProviderShortCode,
-  :createdBy
+  :createdBy,
+  null
 )
-ON CONFLICT (user_keycloak_user_id, permission, data_provider_short_code)
-DO NOTHING
+ON CONFLICT (user_keycloak_user_id, permission, data_provider_short_code) DO UPDATE
```

Comment:
It would seem that this reduces the chance of interleaving queries on this row, but I wasn't confident that it guaranteed no interleaving (e.g. an insert followed by update followed by the insert's do update on conflict), though this part of the postgres documentation suggests exactly one insert or one update will happen:
> ON CONFLICT DO UPDATE guarantees an atomic INSERT or UPDATE outcome; provided there is no independent error, one of those two outcomes is guaranteed, even under high concurrency. This is also known as UPSERT — “UPDATE or INSERT”.

I suppose if there is a concurrent insert/update and delete, it's anybody's guess as to who should win the race anyway.